design/aws-ecs-scheduled-ecs-task-construct.md

```diff
@@ -0,0 +1,141 @@
+# AWS ECS - L3 Construct for Scheduling EC2 Tasks
+
+To address issue [#2352](https://github.com/awslabs/aws-cdk/issues/2352), the ECS CDK construct library should provide a way for customers to create a standalone scheduled task, without creating a service, for their container. The task will be initiated by a cloudwatch event that is scheduled based on a [CW Schedule Event](http://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html).
+
+This would mean adding a new ECS CDK construct `ScheduledEc2Task` and properties `ScheduledEc2TaskProps`, that would take in the necessary properties required to create a Task Definition, an EventRuleTarget as well as an AWS EventRule. To allow for flexibility, container overrides will also be supported properties.
+
+Note: Currently, there is only [CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-reference-ecs.html) support for scheduled tasks on EC2, and not on Fargate. CloudFormation currently only supports clusters, services and taskdefinitions: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-reference-ecs.html.
```

Comment:
This paragraph and the following one are confusing to me in explaining why we're not yet building ScheduledFargateTask.  It's not really about the ECS resource types like Service in CloudFormation, the gap for Fargate and awsvpc network support is in the CFN [CloudWatch Events EcsParameters type](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-events-rule-ecsparameters.html).  In order to support Fargate, I expect the following attributes to be eventually added to EcsParameters in CloudFormation (they already exist in the [CloudWatch Events APIs](https://docs.aws.amazon.com/AmazonCloudWatchEvents/latest/APIReference/API_EcsParameters.html)):
* LaunchType
* NetworkConfiguration (including awsvpcConfiguration with subnets, security groups, etc)
* PlatformVersion
* Group

Reply:
Agreed, I'll update this section. I was leaning more towards adding another ECS resource type that would allow us to create a task (scheduled or otherwise) without a service.

Reply:
Perhaps an open question, but is there utility in creating a task with CFN outside of a service or scheduled task construct? 

If we can accomplish creating a Fargate scheduled task once the CFN EcsParameters object is updated, what use cases exist for creating a 1-off task within a CFN stack? Since they're transient by nature, we'd be creating them with no mechanism to ensure they stick around after the initial stack is spun up.

Reply:
Fair point, updating the doc with the above.